dashboard/test/models/pd/payment_terms_test.rb

```diff
@@ -0,0 +1,52 @@
+require 'test_helper'
+
+class Pd::PaymentTermTest < ActiveSupport::TestCase
+  setup do
+    @regional_partner_1 = create :regional_partner
+    @regional_partner_2 = create :regional_partner
+
+    @workshop_1 = create :pd_workshop, num_sessions: 1, sessions_from: Date.new(2017, 1, 1), regional_partner: @regional_partner_1
+    @workshop_2 = create :pd_workshop, num_sessions: 1, sessions_from: Date.new(2017, 3, 1), regional_partner: @regional_partner_1
+    @workshop_3 = create :pd_workshop, regional_partner: @regional_partner_2
+  end
```

Comment:
For a perf boost, please leverage Will's recently added `transactional_test_case` and `setup_all`. [Example](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/test/controllers/api/v1/pd/workshops_controller_test.rb#L6-L7)